src/site/content/en/notifications/codelab-notifications-service-worker/index.md

```diff
@@ -0,0 +1,201 @@
+
+{% Instruction 'remix', 'ol' %}
+{% Instruction 'preview', 'ol' %}
+
+The Glitch should open in a new Chrome tab.
+
+As you work through this codelab, make changes to the code in the embedded Glitch on this page. Refresh the new tab with your live app to see the changes.
+
+## Get familiar with the sample app and starting code
+
+Start by looking at the live app in the new Chrome tab:
+
+{% Instruction 'devtools-console', 'ol' %}
+
+1. Make sure that the **Info** option is selected in the **Levels** dropdown
+   next to the **Filter** box.
+
+1.  In the DevTools console for your live app, you should see a console message: 
+
+    `TODO: Implement getRegistration()`. 
+
+    This is a message from a function stub that you will implement in this codelab.
+
+Now let's take a look at the sample app's code in the embedded Glitch on this page.
+
+1.  In the embedded Glitch, take a look at `public/index.js`:
+
+    *   There are four stubs for the functions you will implement: `registerServiceWorker`, `getRegistration`, `unRegisterServiceWorker`, and `sendNotification`.
+
+    *   The `requestPermission` function requests the user's permission to send notifications. If you did the [Get started with the Notifications API codelab](/codelab-notifications-get-started), you'll notice that its `requestPermission` function is used here. The only difference is that it now also updates the user interface after resolving the permission request.
+
+    *   The `updateUI` function refreshes all the app's buttons and messages.
+
+    *   The `initializePage` function performs feature detection for service worker capability in the browser and updates the app user interface.
+
+    *   The script waits until the page has loaded and then initializes it.
+
+1.  In the embedded Glitch, open `public/service-worker.js`.
+
+    As the name suggests, you'll add code to the app to register this file as a
+    [service worker](https://developers.google.com/web/fundamentals/primers/service-workers).
+
+    Although the file is not yet in use by the app, it contains some starting code that will print a message to the console when the service worker is activated.
+
+    You'll add code to `public/service-worker.js` to handle notifications when the service worker receives them.
+
+## Register the service worker
+
+In this step, you'll write code that runs
+when the user clicks **Register service worker** in the app UI.
+This code will register `public/service-worker.js` as a service worker.
+
+{% Aside %}
+
+In a real app, you'd just register the service worker automatically from your app's landing page instead of requiring the user to click a button.
+
+For the purposes of this codelab, you'll build an app that you can play with and observe how the service worker interacts with notifications.
+
+{% endAside %}
+
+1.  In the embedded Glitch editor, open `public/index.js`.
+    Replace the `registerServiceWorker` function with the following code:
+
+    ```js
```

Comment:
Indenting the code into the list like this is causing the font size to get bumped by 2px. Normally code is 16px in codelabs, but it's bumping to 18px here.

A quick fix would be to turn these into regular bullets with `*` and not indenting the code.

Reply:
One note: this needs to be an ordered list since a sequence of steps.

Reply:
Yeah I realize it's not ideal to do unordered bullets, but it would fix the issue without requiring a CSS change to the site... I guess we should fix the CSS bug tho :\

Reply:
FWIW, my preference would be to fix the CSS rather than working the content around it. Proposed fix in #1802.